File: SICPA.Models/Entities/Department.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SICPA.Models.Entities
{
    public class Department : Base
    {
        [Key]
        public int Id { get; set; }
        public bool Status { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public string Name { get; set; }
        public string Phone { get; set; }

        [Required]
        [Column("id_enterprise")]
        public int EnterpriseId { get; set; }

        [ForeignKey("EnterpriseId")]
        public Enterprise Enterprise { get; set; }
    }
}

File: SICPA.Angular/src/app/models/department.ts
export class Department {
    id?: number;
    enterpriseId: number;
    name: string;
    description: string;
    phone: string;

    constructor(id: number, enterpriseId: number, name: string, description: string, phone: string)
    {
        this.id = id;
        this.enterpriseId = enterpriseId;
        this.name = name;
        this.description = description;
        this.phone = phone;
    }
}